src/mvba/abba/mod.rs

```diff
@@ -101,6 +103,19 @@ impl Abba {
         }
 
         match &msg.action {
+            Action::Decision(agg_main_vote) => {
+                if let Some(existing_decision) = self.decided_value.as_ref() {
+                    if existing_decision != agg_main_vote {
+                        error!("Existing decision does not match the decision we received: {existing_decision:?} != {agg_main_vote:?}");
+
+                        return Err(Error::Generic("Received conflicting decision".into()));
+                    }
+                    return Ok(());
+                }
+
+                self.decided_value = Some(agg_main_vote.clone());
+                self.broadcast(msg.clone())?; // re-broadcast the msg in case we were the only one who received it.
```

Comment:
Why? This is not consensus responsibility. If we want to make sure other parties will get messages (even they lost) we can have a chose randomly messages and broadcast them

Reply:
It's part of the strong termination spec, as I understand it, the rule is "when a node decides, it broadcasts it's decision."

See the highlight here:
<img width="749" alt="Screen Shot 2022-12-15 at 4 20 09 PM" src="https://user-images.githubusercontent.com/1832378/207840561-4e312d05-04f8-4da8-800c-63e9e8457c87.png">

I think this is to protect against the case where a faulty node only sends the decision to one node. If that one node doesn't broadcast again, then the remaining honest nodes may not terminate (under some worst case secenario).

Reply:
Here we re-broadcast the message.

Reply:
Yeah, isn't that what the highlight says? `send the (aggregated) signature to all parties`

Reply:
I am convinced.